Proposals.Version 2.Iteration 8

shamil-gadelshin opened the pull request:
### Features

- veto for the approved pending execution proposal
- reset a proposal after council reelection
- change upgrade runtime proposal: introduce fixed allowed member list
- add proposals: 'evict storage provider', 'set validator count', 'set storage role parameters'
- set actual proposal type parameters
- set actual proposal runtime parameters
- change 'veto_proposal' to affect approved proposal during the grace period
- fix clippy linter issues
- migrate proposal votes calculations to Perbill 
- remove allowed member_id list for runtime upgrade proposal
- improve comments

shamil-gadelshin committed:
Add codex ProposalDetails for the frontend

shamil-gadelshin committed:
Add ‘create_evict_storage_provider_proposal’ extrinsic

- add ‘create_evict_storage_provider_proposal’ extrinsic to the codex module
- add tests

shamil-gadelshin committed:
Add ‘create_set_validator_count_proposal’ extrinsic

- add ‘create_set_validator_count_proposal’ extrinsic to the codex module
- add tests

shamil-gadelshin committed:
Add ‘create_set_storage_role_parameters_proposal’ extrinsic

- add ‘create_set_storage_role_parameters_proposal’ extrinsic
- add tests

shamil-gadelshin committed:
Fix reset_proposal() in the engine module

- fix reset_proposal() in the engine module: clean VoteExistsByProposalByVoter double map
- add tests

shamil-gadelshin committed:
Change upgrade runtime proposal

- change upgrade runtime proposal in the codex module: add allowed proposers whitelist
- add tests

shamil-gadelshin committed:
Add some comments to proposal modules

shamil-gadelshin committed:
Set min validator count for the proposals

shamil-gadelshin committed:
Add ensure_storage_role_parameters_valid() to the codex

shamil-gadelshin committed:
Add ensure_council_election_parameters_valid() to the codex

shamil-gadelshin committed:
Change min_validator_count() limit in the codex

shamil-gadelshin committed:
Add veto for the pending execution proposal.

shamil-gadelshin committed:
Set general proposal parameters.

shamil-gadelshin committed:
Update proposals runtime parameters

shamil-gadelshin committed:
Set proposal parameters

- set proposal limits and introduce errors
- update spending proposal parameters
- introduce total issuance token percentage

shamil-gadelshin committed:
Fix clippy comments

shamil-gadelshin committed:
Add comments

shamil-gadelshin committed:
Update comments

shamil-gadelshin committed:
Add missing tests

shamil-gadelshin committed:
Add tests for proposal status resolution

shamil-gadelshin committed:
Migrate proposal votes calculations to Perbill

- migrate engine module quorum & threshold calculations from float division to Substrate Perbill type
- add tests

shamil-gadelshin committed:
Remove allowd member_id list for runtime upgrade proposal

bedeho reviewed (changes requested):
- We decided to not have proposal for setting council capacity, for reasons that are too nuanced to unpack here, so just drop that one.

- Nice work on the docs!

shamil-gadelshin committed:
Remove ‘set council mint capacity’ proposal

- remove proposal from the codex
- remove tests
- remove parameters and proposal details
